File: Lab-7/examples/fastapi-scaffold/schemas/agent_state.py
from typing import TypedDict


class AgentState(TypedDict, total=False):
    session_id: str
    user_request: str
    messages: list[dict]
    current_step: str
    approval_required: bool
    approved: bool | None
    retry_count: int
    timeout_ms: int
    last_error: str | None
    research_notes: str | None
    final_response: str | None

File: Lab-7/examples/nextjs-scaffold/lib/agent-state.ts
export interface ChatMessage {
  role: string;
  content: string;
}

export interface AgentState {
  sessionId: string;
  userRequest: string;
  messages: ChatMessage[];
  currentStep: string;
  approvalRequired: boolean;
  approved: boolean | null;
  retryCount: number;
  timeoutMs: number;
  lastError: string | null;
  researchNotes: string | null;
  finalResponse: string | null;
}
